source/thisTarget.ts

```diff
@@ -76,19 +77,41 @@ export function getActionForMessage(
   const isThisTarget = compareTargets(to, thisTarget);
 
   if (!isThisTarget) {
-    debug("The message’s target is", to, "but this is", thisTarget);
+    debug(message.type, "🤫 ignored due to target mismatch", {
+      requestedTarget: to,
+      thisTarget,
+      tabInfoStatus,
+    });
   }
 
   return isThisTarget ? "respond" : "ignore";
 }
 
-let nameRequested = false;
-export async function nameThisTarget() {
-  // Same as above: CS receives messages correctly
-  if (!nameRequested && !thisTarget && !isContentScript()) {
-    nameRequested = true;
-    thisTarget = await messenger("__getTabData", {}, { page: "any" });
-    thisTarget.page = location.pathname + location.search;
+let tabInfoStatus: "needed" | "pending" | "done" | "error" =
+  // The background page already has it
+  isBackground() ||
+  // Same as above: content scripts don't receive broadcasts (yet)
```

Comment:
NIT: ambiguous comment. Does "yet" mean that we'll add support for it later?  That Chrome is adding support for it?

Reply:
I would move this variable closer to `thisTarget`'s definition. This variable describes the intialization status of the  thisTarget definition

Reply:
> There you'll see a mismatch in logic

They're two different things: 

- a target can exist without tab information
- the content script does not have named targets yet

> So the "same as above" part seems wrong

"above" actually referred to this longer comment:

https://github.com/pixiebrix/webext-messenger/blob/fbbc7b166db5ac3243fe35c9aa712cccf55aacec/source/thisTarget.ts#L53-L57

> The comments don't really explain well why contentScript shouldn't have/don't need tabId and frameId information

Indeed the code has a lot of logic developed over time that requires deep understanding of the limitations and behavior of the messaging API. It's impossible to describe the whole logic for every part of the code because it would need to be repeated several times.

I added a `@file` explanation to clarify the intent of the file, when `tabId` is needed, what happens when `tabId` not in the targed, what happens when it's in the target but not ready, etc.